src/standard/statement.ml

```diff
@@ -243,6 +247,10 @@ let assumption ?loc l =
 let check_sat = prove
 let assert_ ?loc t = antecedent ?loc t
 
+let dtype_decl ?loc id pl cstrs =
```

Comment:
Mabe the `dtype_dec` function could take a list of terms (rather than a list of ids) as argument `pl` (this is made so that correct location can be read from the `vars` field) ?